### PR TITLE
feat: add DockerHub builds [VAX-542]

### PR DESCRIPTION
**Pull request by icehaunter**

### CODE REVIEW

File: .buildkite/build_vaxine.yml [hunk 1]
```diff
@@ -9,4 +10,11 @@ agent:
...
     command: make buildkite-build-and-push
-
+  - wait
+  - label: ":rocket: Publish the image to DockerHub"
+    if: build.tag =~ /^[0-9]+\.[0-9]+\.[0-9]+/
+    env:
+      DOCKER_REPO: electricsql
+    command:
+      - export ELECTRIC_IMAGE_NAME="${DOCKERHUB_REPO}/${IMAGE_NAME}"
+      - make make buildkite-build-and-push
```
Comment (define-null) [inline .buildkite/build_vaxine.yml]:
```suggestion
      - make buildkite-build-and-push
```

File: .buildkite/build_vaxine.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 env:
-  DOCKER_REPO: europe-docker.pkg.dev/vaxine/vaxine-io
+  DOCKER_REPO: europe-docker.pkg.dev/vaxine/electric-sql
+  DOCKERHUB_REPO: electricsql
   IMAGE_NAME: vaxine
```
Comment (define-null) [inline .buildkite/build_vaxine.yml:4]: Any benefit in having those envs in a separate file? I would just move all those variables to Makefile together with `export ELECTRIC_IMAGE_NAME="${DOCKERHUB_REPO}/${IMAGE_NAME}"` and would just run single step - `make buildkite-build-and-push`

File: .buildkite/build_vaxine.yml [hunk 3]
```diff
@@ -9,4 +10,11 @@ agent:
 steps:
   - label: ":whale: Build & push the vaxine container"
     command: make buildkite-build-and-push
-
+  - wait
+  - label: ":rocket: Publish the image to DockerHub"
+    if: build.tag =~ /^[0-9]+\.[0-9]+\.[0-9]+/
```
Comment (define-null) [inline .buildkite/build_vaxine.yml:15]: We already take IMAGE_TAG in makefile. Why do we want to be extra protective here?

**Review by define-null — CHANGES_REQUESTED** [review]

File: .buildkite/build_vaxine.yml (hunk 3 above)
Comment (icehaunter) [reply .buildkite/build_vaxine.yml:15]: This second job is run with a different target repository. We don't want to push to DockerHub unless it was from a new Git tag. This also answers why env variables live here - I don't want to mix deployment specifics (i.e. target repos) into Makefile when it's orchestrated by the CI

File: .buildkite/build_vaxine.yml (hunk 1 above)
Comment (icehaunter) [reply .buildkite/build_vaxine.yml]: Good catch

**Review by define-null — APPROVED** [review]

**Pull request merged**